src/server/pkg/obj/obj.go

```diff
@@ -163,16 +164,18 @@ func NewAmazonClientFromSecret(bucket string) (Client, error) {
 			log.Infof("AWS deployed with cloudfront distribution at %v\n", string(distribution))
 		}
 	}
+	// It's ok if we can't find static credentials; we will use IAM roles
+	// in that case.
 	id, err := ioutil.ReadFile("/amazon-secret/id")
-	if err != nil {
+	if err != nil && !os.IsNotExist(err) {
 		return nil, err
 	}
 	secret, err := ioutil.ReadFile("/amazon-secret/secret")
-	if err != nil {
+	if err != nil && !os.IsNotExist(err) {
 		return nil, err
 	}
 	token, err := ioutil.ReadFile("/amazon-secret/token")
-	if err != nil {
+	if err != nil && !os.IsNotExist(err) {
 		return nil, err
 	}
 	region, err := ioutil.ReadFile("/amazon-secret/region")
```

Comment:
Is region required to be present even when the others aren't? If so could we have a comment to that effect?

Reply:
The AWS Go client seems to take a `Region` option, so I'm assuming it's needed for some reason.  I will add a comment.

Reply:
The various AWS regions [have different REST endpoints](http://docs.aws.amazon.com/general/latest/gr/rande.html), so even if authentication is automatic, many clients may still need to be told where to find the AWS APIs. 

There may be some clever way to default this by GETing the magic URL http://169.254.169.254/latest/meta-data and extracting the region from there, but that's probably too clever by half.

Reply:
@emk Thanks for the info.  That's good to know.